src/gurt/debug.c

```diff
@@ -574,16 +574,26 @@ d_log_init_adv(char *log_tag, char *log_file, unsigned int flavor,
 int
 d_log_init(void)
 {
-	char	*log_file;
+	char	*log_file, *log_file_pid_append;
 	int	 flags = DLOG_FLV_LOGPID | DLOG_FLV_FAC | DLOG_FLV_TAG;
 	int	 rc;
+	char     *buffer = NULL;
 
 	log_file = getenv(D_LOG_FILE_ENV);
 	if (log_file == NULL || strlen(log_file) == 0) {
 		flags |= DLOG_FLV_STDOUT;
 		log_file = NULL;
 	}
 
+	log_file_pid_append = getenv(D_LOG_FILE_APPEND_PID_ENV);
+	if (log_file != NULL && log_file_pid_append != NULL) {
+		if (strcmp(log_file_pid_append, "0") != 0) {
+			/* Append pid to log file. */
+			D_ASPRINTF(buffer, "%s%d", log_file, getpid());
```

Comment:
What happens if this allocation fails?

Reply:
Changing to this 
                        /* Append pid to log file. */
                        D_ASPRINTF(buffer, "%s%d", log_file, getpid());
                        if (buffer == NULL)
                                goto out;
                        log_file = buffer;

Reply:
D_GOTO(out, rc = D_NOMEM)